src/fn_call.rs

```diff
@@ -321,7 +325,9 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a+'mir>: crate::MiriEvalContextExt<'a,
                 let ptr = this.read_scalar(args[0])?.not_undef()?;
                 let val = this.read_scalar(args[1])?.to_i32()? as u8;
                 let num = this.read_scalar(args[2])?.to_usize(this)?;
-                if let Some(idx) = this.memory().read_bytes(ptr, Size::from_bytes(num))?.iter().position(
+                if let Some(idx) = this.memory()
+                                       .read_bytes(ptr, Size::from_bytes(num))?
+                                       .iter().position(
```

Comment:
this is worse than before. Maybe move the closure behind the `position(`

Reply:
"Behind"?

Reply:
to the right. idk. there's no good way to clean up the old code. Maybe pull the entire method chain before the `if let` and store it in an aptly named variable.

Reply:
@oli-obk Good idea.